api/agent/agent.go

```diff
@@ -632,10 +635,25 @@ func (a *agent) runHot(ctxArg context.Context, call *call, tok ResourceToken) {
 		stdout: stdoutWrite,
 		stderr: &ghostWriter{inner: stderr},
 	}
-
+	if call.Type == models.TypeApp {
+		// then we'll open a port
+		max := 60000
+		min := 55000
+		cport := rand.Intn(max-min) + min // port for container
+		container.port = strconv.Itoa(cport)
+		call.Format = models.FormatPortHTTP
```

Comment:
we should let docker pick this port dynamically. (eg. publish-all, then find out actual port number)